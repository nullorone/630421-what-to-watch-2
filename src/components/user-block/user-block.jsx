import React from "react";
import PropTypes from "prop-types";

const {string} = PropTypes;

const UserBlock = (props) => {
  const {avatarSrc} = props;

  return (
    <div className="user-block">
      <div className="user-block__avatar">
        <img src={avatarSrc} alt="User avatar" width="63" height="63"/>
      </div>
    </div>
  );
};

UserBlock.propTypes = {
  avatarSrc: string.isRequired,
};

export default UserBlock;
