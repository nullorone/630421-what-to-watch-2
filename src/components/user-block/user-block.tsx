import * as React from "react";

interface UserBlockProps {
  avatarSrc: string;
}

const UserBlock: React.FC<UserBlockProps> = (props) => {
  const {avatarSrc} = props;

  return (
    <div className="user-block">
      <div className="user-block__avatar">
        <img src={avatarSrc} alt="User avatar" width="63" height="63"/>
      </div>
    </div>
  );
};

export default UserBlock;
