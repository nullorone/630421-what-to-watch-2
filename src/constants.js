export const EMPTY_STRING = ``;

export const Value = {
  EMPTY: 0,
  FULL: 1,
};

export const Pathname = {
  DEFAULT: `/`,
  DETAILS: `/details`,
  FILM: `/film`,
};

export const Img = {
  SMALL: {
    width: `280`,
    height: `175`
  },
  BIG: {
    width: `218`,
    height: `327`
  }
};

export const iconNames = [`add`, `fullscreen`, `in_list`, `pause`, `play`];

export const MAX_TIME_LENGTH = 3;

export const MOVIE_NAV_ITEMS = [
  {
    active: false,
    text: `Overview`,
  },
  {
    active: true,
    text: `Details`,
  },
  {
    active: false,
    text: `Reviews`,
  },
];

export const TypeCol = {
  TEXT: `text`,
  REVIEWS: `reviews`,
};

export const AmountSimilarFilms = {
  DEFAULT: 8,
  ON_PAGE_FILM: 4,
};

export const AMOUNT_ADDED_FILMS = 20;

export const MINUTES_IN_HOUR = 60;

export const SECONDS_IN_MINUTE = 60;

export const AMOUNT_COMMENT_IN_COL = 3;

export const MovieNavTabs = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`,
};

export const MILLISECONDS_IN_SECOND = 1000;

export const Url = {
  BASE: `https://htmlacademy-react-2.appspot.com/wtw`,
  FILMS: `/films`,
  PROMO: `/films/promo`,
  LOGIN: `/login`,
  COMMENTS: `/comments`,
  FAVORITE: `/favorite`,
};

export const API_TIMEOUT = 5000;

export const Status = {
  AUTHORIZATION: 401,
  SUCCESS: 200,
};

export const MASK_AVATAR = `/static`;

export const Rating = {
  BAD: `bad`,
  NORMAL: `normal`,
  GOOD: `good`,
  VERY_GOOD: `very good`,
  AWESOME: `awesome`
};
