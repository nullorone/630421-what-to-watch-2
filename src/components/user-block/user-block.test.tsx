import * as React from "react";
import renderer from "react-test-renderer";
import {EMPTY_STRING} from "../../constants";
import {UserBlock} from "./user-block";
import {BrowserRouter} from "react-router-dom";

describe(`Make snapshot`, () => {
  const initProps = {
    avatarSrc: EMPTY_STRING,
    hasAuthorization: true,
  };

  it(`Get snapshot component`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <UserBlock {...initProps}/>
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
