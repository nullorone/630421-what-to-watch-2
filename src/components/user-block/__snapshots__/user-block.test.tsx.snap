// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make snapshot Get snapshot component 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="sign-in.html"
  >
    Sign in
  </a>
</div>
`;
